### PR TITLE
[Event Hubs Client] Track Two (Samples and Read Me)

### PR DESCRIPTION
**Pull request by jsquire**
# Summary

The focus of these changes is to revise the samples and the README content to reflect the latest changes to the client API surface.

# Last Upstream Rebase

Wednesday, November 27, 5:01pm (EST)

# Related and Follow-Up Issues

- [Release Event Hubs Track 2 Library for .NET](https://github.com/Azure/azure-sdk-for-net/issues/8552) (#8552) 
- [Review and update documentation](https://github.com/Azure/azure-sdk-for-net/issues/8553) (#8553) 
- [Create and revise samples](https://github.com/Azure/azure-sdk-for-net/issues/8554) (#8554)

### CODE REVIEW

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample06_PublishAnEventBatchToASpecificPartition.cs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+                // When publishing events, it may be desirable to request that the Event Hubs service place a batch on a specific partition,
+                // for organization and processing.  For example, you may have designated one partition of your Event Hub as being responsible
+                // for all of your telemetry-related events.
+                //
+                // This can be accomplished by setting the identifier of the desired partition when creating the batch.  It is important to note
+                // that if you are using a partition identifier, you may not also specify a partition key; they are mutually exclusive.
+                //
+                // We will publish a small batch of events based on simple sentences.
+
+                // To choose a partition key, you will need to create a custom set of send options.
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample06_PublishAnEventBatchToASpecificPartition.cs]: `partition key` => `partition identifier`

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs [hunk 2]
```diff
@@ -0,0 +1,87 @@
...
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name { get; } = nameof(Sample10_PublishAnEventBatchWithCustomSizeLimit);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description { get; } = "An example of publishing events using a custom size limitation with the batch.";
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs]: Should we replace `{ get; } =` with `=>` like in other samples?

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs [hunk 3]
```diff
@@ -0,0 +1,87 @@
...
+        ///
+        public async Task RunAsync(string connectionString,
+                                   string eventHubName)
+        {
+            // We will start by creating a producer client using its default options.
+
+            await using (var producerClient = new EventHubProducerClient(connectionString, eventHubName))
+            {
+                // There is a limit to the size of an event or batch of events that can be published at once.  This limit varies and depends
+                // on the Event Hubs service and the properties that the namespace of the target Event Hub.  Because the size limit is based
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs]: `depends on the Event Hubs service and the properties that the namespace of the target Event Hub`. Did you mean something like `and the properties of the namespace`?

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_AuthenticateWithClientSecretCredential.cs [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+        ///   The name of the sample.
+        /// </summary>
+        ///
+        public string Name => nameof(Sample11_AuthenticateWithClientSecretCredential);
+
+        /// <summary>
+        ///   A short description of the sample.
+        /// </summary>
+        ///
+        public string Description { get; } = "An example of interacting with an Event Hub using an Azure Active Directory application with client secret for authorization.";
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_AuthenticateWithClientSecretCredential.cs]: `{ get; } =` to `=>` here as well.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_AuthenticateWithClientSecretCredential.cs [hunk 5]
```diff
@@ -0,0 +1,114 @@
...
+                eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Goodbye, Event Hubs!")));
+
+                await producerClient.SendAsync(eventBatch);
+
+                eventsPublished = eventBatch.Count;
+                Console.WriteLine("The event batch has been published.");
+            }
+
+            // With our events published, we'll create a consumer client to read them.  We'll stop reading after we've received all events in the
+            // batch.
```
Comment (kinelski) [inline sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_AuthenticateWithClientSecretCredential.cs]: We are actually reading the from the earliest position in the partition, correct? This is not the point of this sample anyway. Just thought someone might get confused.

**Review by kinelski — DISMISSED** [review]
LGTM.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 6]
```diff
@@ -76,47 +76,40 @@ await using (var client = new EventHubProducerClient(connectionString, eventHubN
...
+    eventBatch.TryAdd(new EventData(Encoding.UTF8.GetBytes("Second")));
     
     await producer.SendAsync(eventsToPublish);
 }
 ```
 
-### Consume events from an Event Hub partition
+### Read events from an Event Hub
+
+In order to consume events from an Event Hub, you'll need to create an `EventHubConsumerClient` for a given consumer group.  When an Event Hub is created, it provides a default consumer group that can be used to get started with exploring Event Hubs.  In our example, we will focus on reading all events that have been published to the Event Hub using an iterator.
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: That was my intent.   Nice catch.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 7]
```diff
@@ -76,47 +76,40 @@ await using (var client = new EventHubProducerClient(connectionString, eventHubN
...
 
-In order to consume events for an Event Hub partition, you'll need to create an `EventHubConsumerClient` for that partition and consumer group combination.  When an Event Hub is created, it provides a default consumer group that can be used to get started.  A consumer also needs to specify where in the event stream to begin receiving events; in our example, we will focus on reading all published events in a partition using an iterator.
+**Note:** It is important to note that this approach to consuming is intended to improve the experience of exploring the Event Hubs client library and prototyping.  It is recommended that it not be used in production scenarios.   For production use, we recommend using the [Event Processor Client](./Azure.Messaging.EventHubs.Processor), as it provides a more robust and performant experience.
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: You're right.  I  missed that one.

File: sdk/eventhub/Azure.Messaging.EventHubs/README.md [hunk 8]
```diff
@@ -201,46 +175,46 @@ For detailed information about these and other exceptions that may occur, please
...
 
-The samples are accompanied by a console application which you can use to execute and debug them interactively.  The simplest way to begin is to launch the project for debugging in Visual Studio or your preferred IDE and provide the Event Hubs connection information in response to the prompts.  
+The samples are accompanied by a console application which you can use to execute and debug them interactively.  The simplest way to begin is to launch the project for debugging in Visual Studio or your preferred editor and provide the Event Hubs connection information in response to the prompts.  
 
 Each of the samples is self-contained and focused on illustrating one specific scenario.  Each is numbered, with the lower numbers concentrating on basic scenarios and building to more complex scenarios as they increase; though each sample is independent, it will assume an understanding of the content discussed in earlier samples.
 
 The available samples are:
 
-- [Hello world](./samples/Sample01_HelloWorld.cs)
+- [Hello world](./samples/Sample01_HelloWorld.cs)
```
[the remark this answers is not in the document]
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/README.md]: That's intentional.   Markdown uses two spaces to denote "please move to a new line"

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs (hunk 2 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs]: Yes, we should.   Copy/paste magic!

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs (hunk 3 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample10_PublishAnEventBatchWithCustomSizeLimit.cs]: Good catch.   That was one heck of an awkward half-phrase.

File: sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_AuthenticateWithClientSecretCredential.cs (hunk 5 above)
Comment (jsquire) [reply sdk/eventhub/Azure.Messaging.EventHubs/samples/Sample11_AuthenticateWithClientSecretCredential.cs]: yeah, it's not ideal if someone is reusing a partition... I kind of shifted away from the "always do latest" to try and reduce complexity and show the intended majority case for `ReadEvents`, though you're right.   I don't have a good answer to this without adding complexity to the text.

**Review by kinelski — APPROVED** [review]
LGTM. No big changes since my last pass.

**Pull request merged**